Update embedded-hal & esp-hal

doraneko94 opened the pull request:
Thank you for creating this valuable crate. Since version `1.0.0` of `embedded-hal` has been released and `esp32-hal` has been deprecated, I have rewritten the crate to use the latest `embedded-hal` and `esp-hal`. Please check it!

doraneko94 committed:
update embedded-hal & esp-hal

rcloran commented:
Hi! Thank you for your PR. I definitely want this sort of change to be part of the project, but don't have time to properly get back into Rust on ESP at the moment.

You could help me by reviewing the other open PR ( #7  ), and I'll ask @KevinBeckers to review this one, and I'd be happy to merge this based on his review if you've also looked at his.

rcloran committed:
Fix example link error

Use println instead of defmt for esp-backtrace.

rcloran committed:
Use more modern build config flags patterns

This moves a flag from `.cargo/config.toml` to `build.rs`, and seems to
apply it only to the example target (ie, avoiding the lib target), and
removes the explicit toolchain from `rust-toolchain.toml` (I guess it
comes from cargo now?).

rcloran committed:
Fix formatting

rcloran commented:
Thank you! Closed in 53b17892d2a596941ce00cbe848fa9e3453fef03
